### PR TITLE
Replace jest with vitest

### PR DESCRIPTION
**Pull request by amanmahajan7**
https://vitest.dev/
https://vitest.dev/guide/migration.html

<del>Blocked on https://github.com/adazzle/react-data-grid/pull/3196</del>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -95,7 +88,8 @@
     "react-router-dom": "^6.3.0",
     "rome": "^11.0.0",
     "typescript": "~4.9.3",
-    "vite": "^4.1.1",
+    "vite": "^^4.1.1",
```
Comment (amanmahajan7) [inline package.json:91]:
```suggestion
    "vite": "^4.1.1",
```

**Comment by nstepien** [comment]
@amanmahajan7 fixed, we just need to re-enable jsdom now I think

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -37,7 +37,8 @@ jobs:
           npm run build
           npm run build:types
       - name: Test
-        run: npm t -- --coverage --colors
+        run: npm t
+        timeout-minutes: 4
```
Comment (amanmahajan7) [inline .github/workflows/ci.yml:41]: Should we keep the timeout?

Comment (nstepien) [reply .github/workflows/ci.yml:41]: Yeah, otherwise it may run for hours

**Review by nstepien — APPROVED** [review]
:shipit:

**Pull request merged**